src/consensus/mod.rs

```diff
@@ -131,21 +141,28 @@ impl FastPaxos {
 
     pub fn step(&mut self, msg: Consensus, view: Vec<&Endpoint>) -> Vec<(Endpoint, Message)> {
         match msg {
-            FastRoundPhase2bMessage(req) => self.handle_fast_round(&req),
+            FastRoundPhase2bMessage(req) => {
+                if let Some(proposal) = self.handle_fast_round(req) {
+                    self.proposal = Some(proposal);
+                }
+            }
             Phase1aMessage(req) => self.paxos.handle_phase_1a(req),
             Phase1bMessage(req) => self.paxos.handle_phase_1b(req),
             Phase2aMessage(req) => self.paxos.handle_phase_2a(req),
             Phase2bMessage(req) => {
-                let proposal = self.paxos.handle_phase_2b(req);
-                // self.on_decide(proposal);
+                if let Some(proposal) = self.paxos.handle_phase_2b(req) {
+                    self.proposal = Some(proposal);
+                }
             }
         };
 
         let mut msgs = Vec::new();
 
         for msg in self.messages.drain(..) {
             match msg {
+                // Unary message
                 (Some(to), msg) => msgs.push((to, msg)),
+                // Broadcast message.. need a better abstraction
```

Comment:
I was thinking we could clean this stuff up later once we get things working.

Reply:
Yup, just a note for the future.